contracts/ExampleToken.cdc

```diff
@@ -106,10 +89,69 @@ pub contract ExampleToken: FungibleToken {
                 ExampleToken.totalSupply = ExampleToken.totalSupply - self.balance
             }
         }
+
+        /// Returns an array with all the views that the ExampleToken implements
+        ///
+        pub fun getViews(): [Type]{
+            return [Type<FungibleTokenMetadataViews.FTView>(),
+                    Type<FungibleTokenMetadataViews.FTDisplay>(),
+                    Type<FungibleTokenMetadataViews.FTVaultData>()]
+                    /*Type<MetadataViews.Medias>()?*/
+                    /*, other views from MetadataViews?? */
+        }
+
+        /// Return the proper struct for the requested Type of view
+        ///
+        pub fun resolveView(_ view: Type): AnyStruct? {
+            switch view {
+                case Type<FungibleTokenMetadataViews.FTView>():
+                    return FungibleTokenMetadataViews.FTView(
+                        ftDisplay: ExampleToken.returnDisplayView(),
+                        ftVaultData: ExampleToken.returnVaultDataView()
+                    )
+                case Type<FungibleTokenMetadataViews.FTDisplay>():
+                    return ExampleToken.returnDisplayView()
+                case Type<FungibleTokenMetadataViews.FTVaultData>():
+                    return ExampleToken.returnVaultDataView()
+            }
+            return nil
+        }
+    }
+
+    // Auxiliary function to return the FTVaultData view
+    access(contract) fun returnVaultDataView(): FungibleTokenMetadataViews.FTVaultData {
+        return FungibleTokenMetadataViews.FTVaultData(
+            storagePath: ExampleToken.VaultStoragePath,
+            publicPath: ExampleToken.ReceiverPublicPath,
+            providerPath: /private/exampleTokenVault,
+            publicLinkedType: Type<&{FungibleToken.Receiver, FungibleToken.Balance, MetadataViews.Resolver}>(),
+            providerLinkedType: Type<&ExampleToken.Vault{FungibleToken.Provider, MetadataViews.Resolver}>(),
+            createEmptyVaultFunction: (fun (): @ExampleToken.Vault {
+                return <-ExampleToken.createEmptyVault()
+            })
+        )
+    }
+
+    // Auxiliary function to return the FTDisplay view
+    access(contract) fun returnDisplayView(): FungibleTokenMetadataViews.FTDisplay {
```

Comment:
Would love to read some thoughts about this functions since this should be focused on "being documentation" and I'm not sure if it brings clarity on how to resolve views. Working on that I just noticed that we do not have an example on how to solve the `NFTView` in the `ExampleNFT` contract